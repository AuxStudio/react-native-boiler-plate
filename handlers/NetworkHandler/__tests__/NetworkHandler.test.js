import React from 'react';
import renderer from 'react-test-renderer';

import { NetworkHandler } from '..';

describe('NetworkHandler', () => {
  let spy;
  const dispatch = jest.fn();

  it('renders with all props', () => {
    expect(renderer.create(<NetworkHandler dispatch={jest.fn()} />)).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(renderer.create(<NetworkHandler dispatch={jest.fn()} />)).toMatchSnapshot();
  });

  it('calls addNetInfoEventListener on componentDidMount', () => {
    spy = jest.spyOn(NetworkHandler.prototype, 'addNetInfoEventListener');

    renderer.create(<NetworkHandler dispatch={dispatch} />);

    expect(spy).toHaveBeenCalled();
  });

  it('calls removeNetInfoEventListener on componentWillUnmount', () => {
    spy = jest.spyOn(NetworkHandler.prototype, 'removeNetInfoEventListener');
    const instance = renderer.create(<NetworkHandler dispatch={dispatch} />);

    instance.unmount();

    expect(spy).toHaveBeenCalled();
  });

  it('updates the store with connection info when WIFI connection', () => {
    const component = renderer.create(<NetworkHandler dispatch={dispatch} />);
    const instance = component.getInstance();
    const connectionInfo = {
      type: 'wifi',
      effectiveType: '4g',
    };

    instance.handleConnectionChange(connectionInfo);
    expect(dispatch).toMatchSnapshot();
  });

  it('updates the store with connection info and goes online when WIFI connection', () => {
    spy = jest.spyOn(NetworkHandler.prototype, 'enableNetwork');
    const component = renderer.create(<NetworkHandler dispatch={dispatch} />);
    const instance = component.getInstance();
    const connectionInfo = {
      type: 'wifi',
      effectiveType: '4g',
    };

    instance.handleConnectionChange(connectionInfo);
    expect(spy).toHaveBeenCalled();
    expect(dispatch).toMatchSnapshot();
  });

  it('updates the store with connection info and goes offline when no connection', () => {
    spy = jest.spyOn(NetworkHandler.prototype, 'disableNetwork');
    const component = renderer.create(<NetworkHandler dispatch={dispatch} />);
    const instance = component.getInstance();
    const connectionInfo = {
      type: 'none',
    };

    instance.handleConnectionChange(connectionInfo);
    expect(spy).toHaveBeenCalled();
    expect(dispatch).toMatchSnapshot();
  });

  it('updates the store with connection info and goes offline when 2g connection', () => {
    spy = jest.spyOn(NetworkHandler.prototype, 'disableNetwork');
    const component = renderer.create(<NetworkHandler dispatch={dispatch} />);
    const instance = component.getInstance();
    const connectionInfo = {
      type: 'cellular',
      effectiveType: '2g',
    };

    instance.handleConnectionChange(connectionInfo);
    expect(spy).toHaveBeenCalled();
    expect(dispatch).toMatchSnapshot();
  });

  afterEach(() => {
    if (spy) {
      spy.mockClear();
    }
  });
});
