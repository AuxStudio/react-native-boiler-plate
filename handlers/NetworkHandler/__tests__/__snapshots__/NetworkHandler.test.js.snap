// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkHandler renders with all props 1`] = `null`;

exports[`NetworkHandler renders with minimum required props 1`] = `null`;

exports[`NetworkHandler updates the store with connection info and goes offline when 2g connection 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "payload": Object {
          "network": Object {
            "effectiveType": "4g",
            "type": "wifi",
          },
        },
        "type": "SET_NETWORK_CONNECTION_INFO",
      },
    ],
    Array [
      Object {
        "type": "enableNetwork",
      },
    ],
    Array [
      Object {
        "payload": Object {
          "network": Object {
            "effectiveType": "4g",
            "type": "wifi",
          },
        },
        "type": "SET_NETWORK_CONNECTION_INFO",
      },
    ],
    Array [
      Object {
        "type": "enableNetwork",
      },
    ],
    Array [
      Object {
        "payload": Object {
          "network": Object {
            "type": "none",
          },
        },
        "type": "SET_NETWORK_CONNECTION_INFO",
      },
    ],
    Array [
      Object {
        "type": "disableNetwork",
      },
    ],
    Array [
      Object {
        "payload": Object {
          "network": Object {
            "effectiveType": "2g",
            "type": "cellular",
          },
        },
        "type": "SET_NETWORK_CONNECTION_INFO",
      },
    ],
    Array [
      Object {
        "type": "disableNetwork",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`NetworkHandler updates the store with connection info and goes offline when no connection 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "payload": Object {
          "network": Object {
            "effectiveType": "4g",
            "type": "wifi",
          },
        },
        "type": "SET_NETWORK_CONNECTION_INFO",
      },
    ],
    Array [
      Object {
        "type": "enableNetwork",
      },
    ],
    Array [
      Object {
        "payload": Object {
          "network": Object {
            "effectiveType": "4g",
            "type": "wifi",
          },
        },
        "type": "SET_NETWORK_CONNECTION_INFO",
      },
    ],
    Array [
      Object {
        "type": "enableNetwork",
      },
    ],
    Array [
      Object {
        "payload": Object {
          "network": Object {
            "type": "none",
          },
        },
        "type": "SET_NETWORK_CONNECTION_INFO",
      },
    ],
    Array [
      Object {
        "type": "disableNetwork",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`NetworkHandler updates the store with connection info and goes online when WIFI connection 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "payload": Object {
          "network": Object {
            "effectiveType": "4g",
            "type": "wifi",
          },
        },
        "type": "SET_NETWORK_CONNECTION_INFO",
      },
    ],
    Array [
      Object {
        "type": "enableNetwork",
      },
    ],
    Array [
      Object {
        "payload": Object {
          "network": Object {
            "effectiveType": "4g",
            "type": "wifi",
          },
        },
        "type": "SET_NETWORK_CONNECTION_INFO",
      },
    ],
    Array [
      Object {
        "type": "enableNetwork",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`NetworkHandler updates the store with connection info when WIFI connection 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "payload": Object {
          "network": Object {
            "effectiveType": "4g",
            "type": "wifi",
          },
        },
        "type": "SET_NETWORK_CONNECTION_INFO",
      },
    ],
    Array [
      Object {
        "type": "enableNetwork",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
